src/python-frontend/python_converter.cpp

```diff
@@ -240,73 +204,85 @@ void python_converter::get_var_assign(
 {
-  // Get variable type
-  current_element_type =
-    get_typet(ast_node["annotation"]["id"].get<std::string>());
-
-  // Id and name
-  std::string name, id;
-  auto target = ast_node["target"];
-  if(!target.is_null() && target["_type"] == "Name")
+  if(ast_node.contains("annotation"))
   {
-    name = target["id"];
-    id = "py:@" + name;
+    // Get type from current annotation node
+    current_element_type =
+      get_typet(ast_node["annotation"]["id"].get<std::string>());
+  }
+  else
+  {
+    // Get type from declaration node
+    std::string var_name = ast_node["targets"][0]["id"].get<std::string>();
+    nlohmann::json ref = find_var_decl(var_name);
+    assert(!ref.empty());
+    current_element_type =
+      get_typet(ref["annotation"]["id"].get<std::string>());
   }
 
-  // Variable location
-  locationt location_begin = get_location_from_decl(ast_node["target"]);
+  exprt lhs;
 
-  // Debug module name
-  // FIXME: This should be read from JSON
-  std::string module_name = "program.py";
+  nlohmann::json value = ast_node["value"];
+  exprt rhs = get_expr(value);
 
-  // Create/init symbol
-  symbolt symbol =
-    create_symbol(module_name, name, id, location_begin, current_element_type);
-  symbol.lvalue = true;
-  symbol.static_lifetime = false;
-  symbol.file_local = true;
-  symbol.is_extern = false;
+  locationt location_begin;
 
-  // Assign a value
-  nlohmann::json value = ast_node["value"];
-  exprt val = get_expr(value);
-  symbol.value = val;
+  if(ast_node["_type"] == "AnnAssign")
+  {
+    // Id and name
+    std::string name, id;
+    auto target = ast_node["target"];
+    if(!target.is_null() && target["_type"] == "Name")
+    {
+      name = target["id"];
+      id = "py:@" + name;
+    }
 
-  code_assignt code_assign(symbol_expr(symbol), symbol.value);
-  code_assign.location() = location_begin;
-  target_block.copy_to_operands(code_assign);
+    // Location
+    location_begin = get_location_from_decl(ast_node["target"]);
 
-  context.add(symbol);
-}
+    // Debug module name
+    std::string module_name = location_begin.get_file().as_string();
 
-exprt python_converter::get_block(const nlohmann::json &ast_block)
-{
-  code_blockt block;
+    // Create/init symbol
+    symbolt symbol = create_symbol(
+      module_name, name, id, location_begin, current_element_type);
+    symbol.lvalue = true;
+    symbol.static_lifetime = false;
+    symbol.file_local = true;
+    symbol.is_extern = false;
+    symbol.value = rhs;
 
-  // Iterate through the statements of the block
-  for(auto &element : ast_block)
-  {
-    StatementType type = get_statement_type(element);
+    lhs = symbol_expr(symbol);
 
-    switch(type)
-    {
-    case StatementType::VARIABLE_ASSIGN:
+    context.add(symbol);
+  }
+  else if(ast_node["_type"] == "Assign")
+  {
+    std::string id = ast_node["targets"][0]["id"].get<std::string>();
+    symbolt *symbol = context.find_symbol(std::string("py:@") + id);
+    if(symbol != nullptr)
```

Comment:
What happens otherwise? Should it even be possible that an assignment to an unknown symbol is being performed and, if so, what's supposed to happen then? I'd suggest to assert this condition.

Reply:
Done